core/src/sumeragi/main_loop.rs

```diff
@@ -313,7 +307,7 @@ impl Sumeragi {
         &mut self,
         block: impl Into<VersionedCommittedBlock>,
     ) {
-        let committed_block = Arc::new(block.into());
+        let committed_block = block.into();
```

Comment:
Why you've decided not to use `Arc` here, it was used to avoid redundant cloning?

Reply:
I avoided redundant clone with other means

Reply:
> Why do you think it will be better?

There is no need for `Arc` in this case. Also, I'd like if we could work toward giving out `&T` instead of `Arc<T>` from `Kura`. It's a bit of a pain point but can be worked around

> Also afaik it wasn't possible to avoid all clones (L347)

that comment is incorrect. I left it before I removed the unnecessary cloning. I believe I have successfully removed all clones

Reply:
> that comment is incorrect 

Indeed, good job then. Cool that it possible to avoid `Arc`ing. Feel free to resolve.